keanu-project/src/main/java/io/improbable/keanu/algorithms/mcmc/nuts/AdaptiveStepSize.java

```diff
@@ -0,0 +1,107 @@
+package io.improbable.keanu.algorithms.mcmc.nuts;
+
+import io.improbable.keanu.algorithms.Statistics;
+import io.improbable.keanu.algorithms.Variable;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+
+import java.util.List;
+
+
+/**
+ * This is based on Algorithm 5 from https://arxiv.org/pdf/1111.4246.pdf.
```

Comment:
Can we mention dual averaging in the comment?

Reply:
👍